spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/profile/AbstractKubernetesProfileEnvironmentPostProcessor.java

```diff
@@ -60,72 +58,64 @@
 		}
 		addNamespaceFromServiceAccountFile(environment);
-		if (isInsideKubernetes(environment)) {
-			if (hasKubernetesProfile(environment)) {
-				if (LOG.isDebugEnabled()) {
-					LOG.debug("'kubernetes' already in list of active profiles");
-				}
-			}
-			else {
-				if (LOG.isDebugEnabled()) {
-					LOG.debug("Adding 'kubernetes' to list of active profiles");
-				}
-				environment.addActiveProfile(KUBERNETES_PROFILE);
-			}
-		}
-		else {
-			if (LOG.isDebugEnabled()) {
-				LOG.warn("Not running inside kubernetes. Skipping 'kubernetes' profile activation.");
-			}
-		}
+		addKubernetesProfileIfMissing(environment);
 	}
 
 	protected abstract boolean isInsideKubernetes(Environment environment);
 
 	private boolean hasKubernetesProfile(Environment environment) {
-		for (String activeProfile : environment.getActiveProfiles()) {
-			if (KUBERNETES_PROFILE.equalsIgnoreCase(activeProfile)) {
-				return true;
-			}
-		}
-		return false;
+		return Arrays.stream(environment.getActiveProfiles()).anyMatch(KUBERNETES_PROFILE::equalsIgnoreCase);
 	}
 
 	@Override
 	public int getOrder() {
 		return ORDER;
 	}
 
-	private void addNamespaceFromServiceAccountFile(ConfigurableEnvironment environment) {
-		Map<String, Object> properties = new HashMap<>();
+	private void addKubernetesProfileIfMissing(ConfigurableEnvironment environment) {
+		if (isInsideKubernetes(environment)) {
+			if (hasKubernetesProfile(environment)) {
+				LOG.debug("'kubernetes' already in list of active profiles");
+			}
+			else {
+				LOG.debug("Adding 'kubernetes' to list of active profiles");
+				environment.addActiveProfile(KUBERNETES_PROFILE);
+			}
+		}
+		else {
+			LOG.warn("Not running inside kubernetes. Skipping 'kubernetes' profile activation.");
+		}
+	}
 
+	private void addNamespaceFromServiceAccountFile(ConfigurableEnvironment environment) {
 		String serviceAccountNamespace = environment.getProperty(NAMESPACE_PATH_PROPERTY,
 				SERVICE_ACCOUNT_NAMESPACE_PATH);
-		LOG.debug("Looking for service account namespace at " + serviceAccountNamespace);
-		boolean serviceAccountNamespaceExists = Files.isRegularFile((new File(serviceAccountNamespace)).toPath(),
-				new LinkOption[0]);
+		LOG.debug("Looking for service account namespace at: [" + serviceAccountNamespace + "].");
+		Path serviceAccountNamespacePath = Paths.get(serviceAccountNamespace);
+		boolean serviceAccountNamespaceExists = Files.isRegularFile(serviceAccountNamespacePath);
 		if (serviceAccountNamespaceExists) {
 			LOG.debug("Found service account namespace at: [" + serviceAccountNamespace + "].");
 
 			try {
-				String namespace = new String(Files.readAllBytes((new File(serviceAccountNamespace)).toPath()));
+				String namespace = new String(Files.readAllBytes((serviceAccountNamespacePath)));
 				LOG.debug("Service account namespace value: " + namespace);
-				properties.put(NAMESPACE_PROPERTY, namespace);
+				environment.getPropertySources().addLast(new MapPropertySource(PROPERTY_SOURCE_NAME,
```

Comment:
this was already present on line `128`, as well as the creation of `new HashMap<>()` at the beginning of the method. I removed both and squashed them in here, the only place where these are needed. A bit less allocations thus, a slight optimization.